docs/requirements.md

```diff
@@ -0,0 +1,51 @@
+| FR3 | When the BMC reports a power status in response to a request to change status (e.g. a power-on is issued), the BMC shall follow through with the request.  This means that if you respond that a power-on event is occuring, that event shall occur. In no case shall the BMC respond that it is completing a power event if it is unable to do so. | We have seen Redfish implementations that exhibit the following incorrect behavior: <ol><li>The power off is issued to all nodes under the BMC.</li><li> Once all nodes have completed the power off (determined via a poll of power status), a power on is issued to all nodes with no wait time.</li><li>If not enough wait time is given, the nodes will complete the posted power on command but not power on.</li></oi>  |
+| FR4 | The BMC when managing a node shall have a means of synchronizing the BMC time to the node time via Redfish. | Many on-node operations require clock time to be set accurately, being able to synchronize via the BMC is very beneficial to the customer. |
+| FR5 | Every ethernet port shall have exactly one mac-address.  In no case shall the BMC and node share the same ethernet port. Each BMC shall have a dedicated ethernet port. | Customer discovery process depends upon MAC address based discovery. Multiplexing MACs on ethernet ports hinders customer discovery. |
+| FR6 | The BMC shall support a means of setting the boot order for the node via Redfish.  The default boot order shall persist across boot attempts. | |
+| FR7 | The BMC shall support communication via HTTP and HTTPS.  | |
+| FR8 | The BMC shall support both sessions and basic-auth as the primary authentication method. | The customer relies almost exclusively on basic-auth. All expectations for performance are predicated on all communication being via basic-auth and not sessions. |
+| FR9 | The node(s) connected to the BMC shall rerun boot options if the boot fails.   It shall not be acceptable for the node to fall into UEFI shell prompt and not retry the other boot options. |  |
+
+## Performance and Quality of Service Requirements
+
+Assumptions:
+1. all communication from the customer to the BMC shall be over HTTP/HTTPS using basic-auth
+2. the customer shall take responsibility to prevent DoS attacks on the system at large. 
+
+|  ID  |  Requirement | Notes  |
+|---|---|---|
+| PR1 | The BMC shall support continious sustained Redfish connections (via basic-auth) of 30 requests per minute. | |
+| PR2 | The BMC shall support a peak load of 100 requests in a single minute (via basic-auth) for a duration of 1 minute, before returning to the continuous sutained load previously defined. | |
+| PR3 | The BMC shall support simultaneous connections via Redfish. | |
```

Comment:
hmm, do you have an idea of a number?

Reply:
No.  Pulling a number out of thin air for CMS I'd say at least 10, maybe more?

This also gets a bit complicated by sessions.   TCP connections is one thing, but we also need some number
of RF sessions to be simultaneously supported.    Again, no idea what's a good number.   This might be a great
topic for next week's Redfish RoundTable.

Reply:
ok for now I set it at 10.  No mention of sessions.  WE should bring up this at the RF roundtable, can you do that? Im  OOO on monday.